Assets/MRTK/Tests/TestUtilities/TestMotionController.cs

```diff
@@ -0,0 +1,147 @@
+        {
+            for (var iter = MoveTo(position + delta, PlayModeTestUtilities.CalculateNumSteps(numSteps)); iter.MoveNext();)
+            {
+                yield return iter.Current;
+            }
+        }
+
+        /// <inheritdoc />
+        public override IEnumerator SetRotation(
+            Quaternion newRotation,
+            int numSteps = PlayModeTestUtilities.ControllerMoveStepsSentinelValue)
+        {
+            Quaternion oldRotation = rotation;
+            rotation = newRotation;
+            yield return PlayModeTestUtilities.SetMotionControllerRotation(
+                oldRotation,
+                newRotation,
+                position,
+                isSelecting,
+                isGrabbing,
+                isPressingMenu,
+                handedness,
+                PlayModeTestUtilities.CalculateNumSteps(numSteps),
+                simulationService);
+        }
+
+        /// <summary>
+        /// Changes the state of the simulated motion controller.
+        /// </summary>
+        /// <param name="isSelecting">Whether the motion controller should be selecting something.</param>
+        /// <param name="isGrabbing">Whether the motion controller should be grabbing something.</param>
+        /// <param name="isPressingMenu">Whether the menu button of the motion controller should be pressed down.</param>
+        /// <param name="waitForFixedUpdate">If true, waits for a fixed update after moving to the new state.</param>
+        public IEnumerator SetState(bool isSelecting, bool isGrabbing, bool isPressingMenu, bool waitForFixedUpdate = true)
+        {
+            for (var iter = PlayModeTestUtilities.MoveMotionController(position, position, isSelecting, isGrabbing, isPressingMenu, handedness, simulationService, 1); iter.MoveNext();)
+            {
+                yield return iter.Current;
+            }
+            if (waitForFixedUpdate)
+            {
+                yield return new WaitForFixedUpdate();
+            }
+        }
+
+        /// <summary>
+        /// Combined sequence of selecting and unselecting
+        /// </summary>
+        public override IEnumerator Click()
+        {
+            yield return SetState(true, false, false);
+            yield return null;
+            yield return SetState(false, false, false);
+            yield return null;
+        }
+
+        /// <summary>
+        /// Combined sequence of selecting, moving, and releasing.
+        /// </summary>
+        /// <param name="positionToRelease">The position to which the hand moves while pinching</param>
+        /// <param name="waitForFinalFixedUpdate">Wait for a final physics update after releasing</param>
+        /// <param name="numSteps">Number of steps of the hand movement</param>
+        public IEnumerator SelectAndThrowAt(Vector3 positionToRelease, bool waitForFinalFixedUpdate, int numSteps = 30)
```

Comment:
Instead of using the fixed 30, should we use PlayModeTestUtilities.ControllerMoveStepsSentinelValue?

Reply:
I don't think so as ControllerMoveStepsSentinelValue may be translated into two values depending on the situation but both are different from the value here (also the signature is based on the existing TestHand)